frontend/src/components/admin/DatePicker.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { Text, Stack, Select, Input } from "@chakra-ui/react";
+
+type DatePickerProps = {
+  label: string;
+  month: string;
+  year: string;
+  setMonth: (month: string) => void;
+  setYear: (year: string) => void;
+  invalid?: boolean;
+};
+
+const DatePicker = ({
+  label,
+  month,
+  year,
+  setMonth,
+  setYear,
+  invalid,
+}: DatePickerProps): React.ReactElement => {
+  enum Months {
+    January = "01",
+    February = "02",
+    March = "03",
+    April = "04",
+    May = "05",
+    June = "06",
+    July = "07",
+    August = "08",
+    September = "09",
+    October = "10",
+    November = "11",
+    December = "12",
+  }
+
+  const handleYearInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setYear(event.target.value.replace(/\D/g, ""));
+  };
+
+  const handleMonthInput = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setMonth(Months[event.target.value as keyof typeof Months]);
+  };
+
+  return (
+    <Stack direction="column">
+      <Text>{label}</Text>
```

Comment:
Could you use the FormLabel and FormControl elements for this by any chance? It will have better accessibility properties and will make focusing the inputs easier.

Reply:
Sure, I'm not really sure what props I should add for accessibility though. Should I be adding aria-labels?

Reply:
Should be fine to just use FormLabel and FormControl, Chakra components should already have any necessary aria- attributes specified.